tests/fault_tolerance/deploy/validations.py

```diff
@@ -0,0 +1,981 @@
+                            if count > 0:
+                                logger.info(
+                                    f"  - Container '{container_name}': {count} restart(s)"
+                                )
+                        scenario_verified = True
+                    else:
+                        logger.warning(
+                            f"⚠ No container restarts detected for {pod_name}. "
+                            f"Process may not have been killed or restart was too fast."
+                        )
+                else:
+                    logger.warning(f"Could not get restart count for {pod_name}")
+
+                # Method 2: Check for restart events
+                logger.info("\nChecking K8s events for container restarts...")
+                found_events = check_container_restart_events(
+                    deployment, pod_name, namespace
+                )
+                if found_events:
+                    logger.info(f"✓ Container restart events found for {pod_name}")
+                    if not scenario_verified:
+                        scenario_verified = True
+
+            if scenario_verified:
+                logger.info("\n✓ STAGE 1.1 PASSED: Process termination confirmed")
+            else:
+                logger.warning(
+                    "\n⚠ STAGE 1.1 WARNING: Could not fully confirm process termination. "
+                    "This may be OK if container restart was very fast."
+                )
+        else:
+            logger.warning("No process termination failures found in affected_pods")
+    else:
+        logger.info("Skipping process termination verification (missing required info)")
+        logger.info(
+            "Note: For process termination scenarios, validation focuses on results (Stage 2)"
+        )
+
+    logger.info("")
+    return scenario_verified
+
+
+# ============================================================================
+# Results Verification Functions (Stage 2)
+# ============================================================================
+
+
+def verify_results_high_availability(
+    scenario: Scenario,
+    metrics: Dict[str, Any],
+    recovery_time: Optional[float] = None,
+    min_success_rate: Optional[float] = 0.90,
+    max_recovery_time: Optional[float] = 60,
+    **kwargs,
+) -> None:
+    """Verify results for high-availability scenarios (with redundancy).
+
+    Validates:
+    - High success rate (>90%)
+    - Fast recovery time (<60s)
+    - System handled failures gracefully
+
+    Args:
+        scenario: Scenario object
+        metrics: Parsed metrics from results
+        recovery_time: Recovery time in seconds
+        min_success_rate: Minimum acceptable success rate
+        max_recovery_time: Maximum acceptable recovery time
+        **kwargs: Additional arguments
+
+    Raises:
+        AssertionError: If validation fails
+    """
+    logger.info("╔" + "═" * 78 + "╗")
```

Comment:
there seems to be a lot of similar code in the logging and the specific "stage" - is there a way to simplify and /or remove some of these?